### PR TITLE
D3D: Implement Z pokes.

### PR DESCRIPTION
**Pull request by Armada651**
This also fixes an offset issue with color pokes.

This is just a quick implementation of Z pokes. It should still be expanded to support the EFB poke method used in the OGL backend.

### CODE REVIEW

File: Source/Core/VideoBackends/D3D/D3DUtil.cpp [hunk 1]
```diff
@@ -572,19 +572,19 @@ void drawShadedTexQuad(ID3D11ShaderResourceView* texture,
...
+		{ x2, y1, z, Color },
+		{ x1, y2, z, Color },
+		{ x2, y2, z, Color },
 	};
 
 	if (cq_observer ||
 	    draw_quad_data.x1 != x1 || draw_quad_data.y1 != y1 ||
 	    draw_quad_data.x2 != x2 || draw_quad_data.y2 != y2 ||
-	    draw_quad_data.col != Color)
+		draw_quad_data.col != Color || draw_quad_data.z != z)
```
Comment (BhaaLseN) [inline Source/Core/VideoBackends/D3D/D3DUtil.cpp]: Alignment looks a bit off in the GitHub diff, can you check this again? :S

**Comment by JMC47** [comment]
Yay.

**Comment by JMC47** [comment]
Verified to work in Tony Hawk 4, That 007 game, and Monster Hunter Tri.

**Comment by Armada651** [comment]
Um, I actually wanted to wait until I had time to implement EFB pokes
properly. But if it's this urgent then I guess it's okay.
Op 8 jun. 2015 5:25 AM schreef "Ryan Houdek" <notifications@github.com>:

**Comment by Sonicadvance1** [comment]
No time like the present.

**Pull request merged**